From: Thorsten Blum <thorsten.blum@toblux.com>
Subject: lib: test_hmm: use min() to improve dmirror_exclusive()
Date: Fri, 26 Jul 2024 15:12:46 +0200

Use min() to simplify the dmirror_exclusive() function and improve its
readability.

Link: https://lkml.kernel.org/r/20240726131245.161695-1-thorsten.blum@toblux.com
Signed-off-by: Thorsten Blum <thorsten.blum@toblux.com>
Cc: Jérôme Glisse <jglisse@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/test_hmm.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/lib/test_hmm.c~lib-test_hmm-use-min-to-improve-dmirror_exclusive
+++ a/lib/test_hmm.c
@@ -799,10 +799,7 @@ static int dmirror_exclusive(struct dmir
 		unsigned long mapped = 0;
 		int i;
 
-		if (end < addr + (ARRAY_SIZE(pages) << PAGE_SHIFT))
-			next = end;
-		else
-			next = addr + (ARRAY_SIZE(pages) << PAGE_SHIFT);
+		next = min(end, addr + (ARRAY_SIZE(pages) << PAGE_SHIFT));
 
 		ret = make_device_exclusive_range(mm, addr, next, pages, NULL);
 		/*
_
